### PR TITLE
add rocksdb 6.8.1

### PR DESCRIPTION
**Pull request by gocarlos**
took another try on rocksdb from here:  https://github.com/conan-io/conan-center-index/pull/410

here I tried the latest version e.g. master as it has several cmake fixes regarding shared libs etc. though its blocked by https://github.com/facebook/rocksdb/issues/6725

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/rocksdb/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,131 @@
+from conans import ConanFile, CMake, tools
+from conans.tools import Version
+from conans.errors import ConanInvalidConfiguration
+import os
+
+
+class RocksDB(ConanFile):
+    name = "rocksdb"
+    homepage = "https://github.com/facebook/rocksdb"
+    license = "GPL-2.0"
```
Comment (Croydon) [inline recipes/rocksdb/all/conanfile.py:10]:
```suggestion
    license = ("GPL-2.0-only", "Apache-2.0")
```

https://github.com/facebook/rocksdb#license

**Comment by syoliver** [comment]
Please find a proposal to improve this PR here [rocksdb/8.6.1](https://github.com/gocarlos/conan-center-index/pull/6)
* I fixed an issue with MSVC shared version
* I removed the static rocksdb libraries for shared variant. I was hidding previous issue.
* I moved the version to 8.6.1 which seems to fix your error with gtest (at least I didn't reproduce it)

**Comment by syoliver** [comment]
Oups I forgot to push my last commit... Sorry, the fix is trivial but I can only push it tonight if you don't fix it before...

**Comment by syoliver** [comment]
Plase find my previous missing commit here : [#7](https://github.com/gocarlos/conan-center-index/pull/7)

**Comment by gocarlos** [comment]
> Plase find my previous missing commit here : [#7](https://github.com/gocarlos/conan-center-index/pull/7)

great, thanks, you can also use github suggestions, easier than merge requests I think :+1:

File: recipes/rocksdb/all/test_package/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    settings = "os", "compiler", "build_type", "arch"
+    generators = "cmake"
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.definitions["ROCKSDB_DLL"] = self.options["rocksdb"].shared
+
+        cmake.configure()
+        cmake.build()
+
```
Comment (syoliver) [inline recipes/rocksdb/all/test_package/conanfile.py:15]:
```suggestion

    def config_options(self):
        minimal_cpp_standard = "11"
        if self.settings.compiler.cppstd:
            tools.check_min_cppstd(self, minimal_cpp_standard)

```

Would it make sense, as you did in the conanfile.py ?

File: recipes/rocksdb/all/test_package/test_package_stable_abi.cpp [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+int main() {
+  rocksdb_options_t *options = rocksdb_options_create();
+  rocksdb_options_set_create_if_missing(options, 1);
+  char* err = NULL;
+  rocksdb_t* db = rocksdb_open(options, "testdb_stable", &err);
+
+  if (!db) {
+    printf("DB error: %s\n", err);
+  } else {
+    rocksdb_free(db);
```
Comment (syoliver) [inline recipes/rocksdb/all/test_package/test_package_stable_abi.cpp:14]:
```suggestion
    rocksdb_close(db);
```

File: recipes/rocksdb/all/test_package/test_package.cpp [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+int main() {
+  rocksdb::DB* db;
+  rocksdb::Options options;
+  options.create_if_missing = true;
+  rocksdb::Status status = rocksdb::DB::Open(options, "testdb", &db);
+  
+  if (!status.ok()) {
+    std::cerr << "DB error: " << status.ToString() << std::endl;
+  }
+
```
Comment (syoliver) [inline recipes/rocksdb/all/test_package/test_package.cpp:14]:
```suggestion
  delete db;
```

**Comment by syoliver** [comment]
Actually, there is a new version : 6.10.1
But I would prefer merging this PR and add another PR for the new version...

File: recipes/rocksdb/all/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+            self._remove_cpp_headers() # Force stable ABI for shared libraries
+
+        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
+
+    def package_info(self):
+        self.cpp_info.names["cmake_find_package"] = "RocksDB"
+        self.cpp_info.names["cmake_find_package_multi"] = "RocksDB"
+        self.cpp_info.libs = tools.collect_libs(self)
+        if self.settings.os == "Windows":
+            self.cpp_info.system_libs = ["Shlwapi.lib", "Rpcrt4.lib"]
```
Comment (danimtb) [inline recipes/rocksdb/all/conanfile.py:151]: I think the ".lib" extensions are not needed
```suggestion
            self.cpp_info.system_libs = ["Shlwapi", "Rpcrt4"]
```

**Comment by ZaMaZaN4iK** [comment]
Can we merge it?

**Pull request merged**